### DISCUSSION TITLE
Create verification token if verification is required

### DISCUSSION THREAD

**Discussion by ramses-jordan** [fastapi-users/fastapi-users#604] [Q&A] [answered]
If you require verification for a user when creating the router, you have to then call the `/request-verify-token` endpoint which isn't practical as it should generate a token after verification. How I currently solve this is like in the example below:

```python
from fastapi_users.router.verify import VERIFY_USER_TOKEN_AUDIENCE
from fastapi_users.utils import generate_jwt
from app.config import settings

def generate_verification_token(user: UserDB) -> str:
    token_data = {
        "user_id": str(user.id),
        "email": str(user.email),
        "aud": VERIFY_USER_TOKEN_AUDIENCE,
    }
    token = generate_jwt(
        token_data,
        settings.VERIFICATION_SECRET,
        3600,
    )
    return token

def on_after_register(user: UserDB, request: Request):
   token = generate_verification_token(user)
   print(f"{settings.FRONTEND_URL}/verify/?token={token}"

fastapi_users.get_register_router(on_after_register)
fastapi_users.get_users_router(requires_verification=True)
```

Is there a better way to directly generate a verification token?

**Comment by frankie567** [comment, marked as answer]
Hi there!

By design, the logic for verification and registration is clearly separated. This is why you need to call the `/request-verify-token` endpoint after `/register`. Is that such a big problem? 

If you really need to do it in one call, I think what you did is the right way to go.

**Comment by bert2002** [comment]
Please note since 10.3.0 you need to use `sub` instead of `user_id`. E.g.
```
    token_data = {
        "sub": str(user.id),
        "email": str(user.email),
        "aud": VERIFY_USER_TOKEN_AUDIENCE,
    }
```
